### ISSUE TITLE
Wrong type detection TValue

### ISSUE THREAD

**Issue by AbyssMorganCTI** [DEVSENSE/phptools-docs#503]
![image](https://github.com/DEVSENSE/phptools-docs/assets/72556823/7734b4b0-aac3-482b-87b6-877e6f535062)

```php
public function getAttributes(array $config) : Collection

$attributes = $macro->getAttributes($config);

public function bindAttribute(FileAttribute $definition, mixed $value) : string
```

Stopped working after last update.

**Comment by jakubmisek** [comment]
Thank you for reporting the issue. I have tried a quick test case (using `"laravel/framework": "^10.10"`), but it works in my case :/

```php
class ATest
{
    function test() {
        $attrs = $this->getAttributes();
        foreach ($attrs as $attr) {
            $this->bindAttribute($attr);
        }
    }

    function getAttributes(): Collection {
        return User::whereIn('name', [])->get();
    }

    function bindAttribute(User $u) { /*...*/ }
}
```

Is there anything I missed? The `TValue` must be passed through the code in a way we don't handle correctly.

**Comment by AbyssMorganCTI** [comment]
Your code looks similar to what I have.

Laravel Framework 10.42.0

On changelog i see "Improved foreach type inferring when taking advantage of generic type arguments and _ide_helper."

composer.json
```json
{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": ["framework", "laravel"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": ">=8.1.0",
        "barryvdh/laravel-dompdf": "v2.0.1",
        "doctrine/dbal": "^3.0",
        "dragonbe/vies": "v2.2.0",
        "google/auth": "^1.0",
        "gusapi/gusapi": "v5.3.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "jurosh/pdf-merge": "v2.1.0",
        "laravel/framework": "^10.0",
        "laravel/tinker": "^2.0",
        "laravel/ui": "^4.0",
        "liborm85/composer-vendor-cleaner": "^1.7",
        "maatwebsite/excel": "^3.1",
        "phpoffice/phpspreadsheet": "^1.18",
        "phpoffice/phpword": "0.18.3",
        "rap2hpoutre/fast-excel": "^5.3",
        "serwer/sms": "dev-master",
        "spatie/calendar-links": "v1.4.3",
        "spatie/laravel-ignition": "^2.0",
        "symfony/error-handler": "^6.3",
        "symfony/mailer": "^6.3",
        "twilio/sdk": "^6.44",
        "tymon/jwt-auth": "^2.0"
    },
    "require-dev": {
        "filp/whoops": "2.15.4",
        "mockery/mockery": "1.3.4",
        "nunomaduro/collision": "^7.0",
        "phpunit/phpunit": "^10.0"
    },
    "autoload": {
        "classmap": [
            "database/seeds",
            "database/factories"
        ],
        "psr-4": {
            "App\\": "app/"
        }
    },
    "autoload-dev": {
        "psr-4": {
          "App\\": "app/",
          "Database\\Factories\\": "database/factories/",
          "Database\\Seeders\\": "database/seeders/"
        }
    },
    "scripts": {
        "post-create-project-cmd": [
            "@php artisan key:generate"
        ],
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true,
        "platform-check": false,
        "allow-plugins": {
            "liborm85/composer-vendor-cleaner": true
        }
    },
    "minimum-stability": "stable",
    "prefer-stable": true
}
```

**Comment by jakubmisek** [comment]
Thank you,

Do you have generated ide_helper or laravel helper files? They might contain some doc comments we don't handle as expected.

**Comment by jakubmisek** [comment]
May I ask about the `$macro->getAttributes($config);` ?

$macro is an object, and `getAttributes` is declared in a trait or a mixin?

**Comment by ging-dev** [comment]
> May I ask about the `$macro->getAttributes($config);` ?
> 
> $macro is an object, and `getAttributes` is declared in a trait or a mixin?

It may be related to override/implement:

```php
<?php

/**
 * @template T
 */
interface A
{
    /**
     * @param T $name
     * @return iterable<T>
     */
    public function test($name);
}

/**
 * @implements A<SplFileInfo>
 */
class B implements A {
    public function test($name) {
    }
}

$test = (new B)->test('hello'); // Argument '1' passed to test() is expected to be of type T, string given

<?php

/**
 * @template TValue
 */
class A {
    /**
     * @param TValue $name
     * @return TValue
     */
    public function test($name) {

    }
}

/**
 * @extends A<string>
 */
class B extends A {
    public function test($name) {

    }
}

(new B)->test(1); // Argument '1' passed to test() is expected to be of type TValue, int given
```

**Comment by jakubmisek** [comment]
Thanks!

**Comment by jakubmisek** [comment]
Fixed in `v1.45.15192`, thanks!
